storage-api/src/java/org/apache/hadoop/hive/common/AcidMetaDataFile.java

```diff
@@ -55,7 +55,9 @@ public String toString() {
     // written by Major compaction
     COMPACTED,
     // written by truncate
-    TRUNCATED;
+    TRUNCATED,
+    // written by drop partition
+    DROPPED;
```

Comment:
Will this cause backward incompatibility? What will happen in the old code trying to read this file?

Reply:
it's hard to be backward compatible with such a strict code. Not sure why we need to throw an exception here. Most probably we would need to upgrade all HMS instances.   
````
String dataFormat = metaData.getOrDefault(Field.DATA_FORMAT, "null");
        switch (dataFormat) {
          case Value.COMPACTED:
            return true;
          default:
            throw new IllegalArgumentException("Unexpected value for " + Field.DATA_FORMAT
                + ": " + dataFormat);
        }
````

Reply:
Could we make it more forgiving?
Maybe allowing for some default behaviour?

Reply:
ok, looks like metafile is a legacy, prior v4.0 code, so we could omit it.

Reply:
it's ok now, however older versions would have to be upgraded